### PR TITLE
[MBL-17263][Student] Create, edit, delete To Do

### PR DESCRIPTION
**Pull request by kristofnemere**
Test plan: 
- same functionality as in the Flutter version
- slightly different design (check Figma)

refs: MBL-17263
affects: Student
release note: none

## Checklist

- [x] Follow-up e2e test ticket created or not needed
- [x] Tested in dark mode
- [x] Tested in light mode
- [x] A11y checked

### CODE REVIEW

**Comment by kristofnemere** [comment]
e2e: https://app.bitrise.io/build/34477769-450a-4ebc-94da-db90a37ef128

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Review by kdeakinstructure — CHANGES_REQUESTED** [review]
I've written my findings to you via Slack

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Review by kdeakinstructure — APPROVED** [review]
QA 👍

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

File: libs/pandautils/src/main/java/com/instructure/pandautils/features/todo/createupdate/CreateUpdateToDoViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+    private val repository: CreateUpdateToDoRepository
+) : ViewModel() {
+
+    private val _uiState = MutableStateFlow(CreateUpdateToDoUiState())
+    val uiState = _uiState.asStateFlow()
+
+    private val _events = Channel<CreateUpdateToDoViewModelAction>()
+    val events = _events.receiveAsFlow()
+
+    private val initialDate = savedStateHandle.get<String>(CreateUpdateToDoFragment.INITIAL_DATE).toSimpleDate()?.toLocalDate() ?: LocalDate.now()
```
Comment (tamaskozmer) [inline libs/pandautils/src/main/java/com/instructure/pandautils/features/todo/createupdate/CreateUpdateToDoViewModel.kt:60]: You can pass a `clock` parameter to the `.now` function to make testing more felxible. (optional)

File: libs/pandautils/src/main/java/com/instructure/pandautils/features/todo/createupdate/composables/SelectCalendarScreen.kt [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+                                Box(
+                                    modifier = Modifier
+                                        .size(24.dp)
+                                        .clip(CircleShape)
+                                        .background(color = Color(course.backgroundColor))
+                                )
+                                Text(
+                                    text = course.name,
+                                    fontSize = 16.sp,
+                                    fontWeight = FontWeight(600),
```
Comment (tamaskozmer) [inline libs/pandautils/src/main/java/com/instructure/pandautils/features/todo/createupdate/composables/SelectCalendarScreen.kt:162]: I think these font weighs are causing the design issues I mentioned in the QA findings. I know it's in Figma but seems like it's wrong.

File: libs/pandautils/src/main/java/com/instructure/pandautils/features/todo/createupdate/composables/CreateUpdateToDoScreen.kt [hunk 3]
```diff
@@ -0,0 +1,551 @@
...
+): TimePickerDialog {
+    val calendar = Calendar.getInstance()
+    calendar.set(Calendar.HOUR_OF_DAY, uiState.time.hour)
+    calendar.set(Calendar.MINUTE, uiState.time.minute)
+    return TimePickerDialog(
+        context,
+        { _, hourOfDay, minute ->
+            actionHandler(
+                CreateUpdateToDoAction.UpdateTime(
+                    LocalTime.of(
```
Comment (tamaskozmer) [inline libs/pandautils/src/main/java/com/instructure/pandautils/features/todo/createupdate/composables/CreateUpdateToDoScreen.kt:510]: You can use `toLocalTime` extension here.

File: libs/pandautils/src/main/java/com/instructure/pandautils/features/todo/createupdate/composables/CreateUpdateToDoScreen.kt [hunk 4]
```diff
@@ -0,0 +1,551 @@
...
+    actionHandler: (CreateUpdateToDoAction) -> Unit
+): DatePickerDialog {
+    val calendar = Calendar.getInstance()
+    calendar.set(uiState.date.year, uiState.date.monthValue - 1, uiState.date.dayOfMonth)
+    return DatePickerDialog(
+        context,
+        { _, year, month, dayOfMonth ->
+            actionHandler(
+                CreateUpdateToDoAction.UpdateDate(
+                    LocalDate.of(
```
Comment (tamaskozmer) [inline libs/pandautils/src/main/java/com/instructure/pandautils/features/todo/createupdate/composables/CreateUpdateToDoScreen.kt:478]: You can use `toLocalDate` extension here.

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

File: libs/pandautils/src/main/java/com/instructure/pandautils/features/calendar/composables/CalendarScreen.kt [hunk 5]
```diff
@@ -124,6 +126,7 @@ fun CalendarScreen(
                 Surface(
                     modifier = Modifier
                         .fillMaxSize()
+                        .padding(padding)
```
[the remark this answers is not in the document]
Comment (kristofnemere) [reply libs/pandautils/src/main/java/com/instructure/pandautils/features/calendar/composables/CalendarScreen.kt:133]: I think that padding param from content should be passed everytime

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Review by tamaskozmer — CHANGES_REQUESTED** [review]

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Review by kdeakinstructure — APPROVED** [review]
QA 👍

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Pull request merged**